types/varint/mod.ts

```diff
@@ -0,0 +1,40 @@
+import { Type } from "../types.ts";
+
+const SEGMENT_BITS = 0x7F;
+const CONTINUE_BIT = 0x80;
+
+export class Leb128Varint implements Type<number> {
+  read(dataView: DataView, byteOffset = 0): number {
+    let value = 0, position = 0;
+    while (true) {
+      const currentByte = dataView.getInt8(byteOffset);
+      value |= (currentByte & SEGMENT_BITS) << position;
+
+      if ((currentByte & CONTINUE_BIT) === 0) break;
+
+      position += 7;
+      byteOffset++;
+
+      if (position >= 32) {
+        throw new Error("VarInt is too big");
```

Comment:
```suggestion
        throw new TypeError("VarInt is too big");
```

Perhaps a more descriptive error would be nice, `Leb128Varint cannot exceed 32 bits in length` perhaps?